includes/class-wc-gateway-stripe.php

```diff
@@ -1324,4 +1333,30 @@ private function validate_field( $field_key, $field_value ) {
 
 		return $this->validate_text_field( $field_key, $field_value );
 	}
+
+	/**
+	 * Migrate the card setting on Stripe gateway activation/upgrade if it's not set.
+	 *
+	 * In WooCommerce Stripe 8.0 the plugin settings were redesigned and whether this gateway is
+	 * enabled no longer solely represents whether the card method is enabled. After upgrading to 8.0,
+	 * migrate the settings so cards are enabled depending on whether this gateway was enabled.
+	 *
+	 * @since 8.0.0
+	 */
+	public function maybe_migrate_the_card_setting() {
+		$previous_version = get_option( 'wc_stripe_version' );
+
+		// Cards are enabled by default.
+		if ( ! $previous_version ) {
+			$this->update_option( 'cards_enabled', 'yes' );
+		}
+		// Exit if it's a new install or the previous version is already 7.8.0 or greater.
+		if ( version_compare( $previous_version, '8.0.0', '>=' ) ) {
+			return;
+		}
+
+		if ( ! $this->get_option( 'cards_enabled' ) ) {
+			$this->update_option( 'cards_enabled', $this->enabled );
+		}
```

Comment:
Because the `cards_enabled` option is new, upon updating to 8.0 I make sure to set this new option to what was the old approach. 

ie on 7.3 and prior, if you enabled the main gateway, you enabled cards and so after updating to 8.0, I make sure to set this option based on whether the main gateway was enabled. 

I haven't tested this part, but it should theoretically work.